### ISSUE TITLE
Update consul adapter yaml to use the latest version of consul

### ISSUE THREAD

**Issue by girishranganathan** [layer5io/meshery-consul#2]
**Describe the issue**
Consul adapter was created right before KubeCon EU 2019. After that we have not updated the adapter to use the latest updates in Consul.

The file which will need update is https://github.com/layer5io/meshery-consul/blob/master/consul/config_templates/consul.yaml

**Comment by mgfeller** [comment]
Looks like this file was generated using Helm. 
Which value file was used?

**Comment by leecalcote** [comment]
@Blake might some insight here.

**Comment by mgfeller** [comment]
I can figure it out. Although, is it important to look at the old parametrization?

Do we have some guidelines for the parametrization of the meshes that are deployed?
For instance, should they be production ready, or is ease of exploring (e.g. working through tutorials) better suited? Should it be based on (one of) the default/example configs of the distribution?

Also, could a requirement be that a mesh can be deployed to one node (see #1 )?

**Comment by mgfeller** [comment]
### Testing

* deploy Consul Connect to a cluster, use a namespace other than default or consul
* the Consul UI is activated, access it and check the Consul version in the footer, it should be current, >= 1.8.0
* alternatively, use `kubectl get -n <namespace> statefulset/consul-consul-server -o jsonpath='{.spec.template.spec.containers[0].image}'` to get the consul image and its version
* deploy an app to the cluster that has connect-inject enabled, e.g. image-hub, and communicates with backend services in the cluster
* verify it shows up in the Consul UI, under Services
* access this app, verify it works
* Create a deny-all intention using the Consul GUI
* verify that the app's functionality is reduces due to the fact that backend services cannot be reached anymore (couldn't understand how this should manifest itself in the image-hub application)
* delete the app from the cluster
* delete Consul from the cluster, verify using kubectl that it has been deleted.

**Comment by blake** [comment]
> Do we have some guidelines for the parametrization of the meshes that are deployed?
For instance, should they be production ready, or is ease of exploring (e.g. working through tutorials) better suited? Should it be based on (one of) the default/example configs of the distribution?

If you're looking for a basic config that can be deployed on one host (not recommended for prod) with most production features enabled, then you can use something like this.

1. Generate an encryption key to secure Consul's gossip communication.
    `kubectl create secret generic consul-gossip-encryption-key --from-literal=key=$(consul keygen)`
2. Create `values.yaml` file configure the Helm chart deployment.

    ```yaml
    ---
    # Example config for Meshery
    # The list of supported values can be found at https://www.consul.io/docs/k8s/helm.html
    global:
      # Enable ACLs
      acls:
        manageSystemACLs: true

      # Enable mTLS between Consul clients and servers
      tls:
        enabled: true
        enableAutoEncrypt: true

      # Secure gossip communication with pre-defined key
      gossipEncryption:
        secretName: consul-gossip-encryption-key
        secretKey: key

    # Enable sidecar injection for annotated pods
    connectInject:
      enabled: true

    server:
      # Install Consul with a single server node
      replicas: 1
      bootstrapExpect: 1
    ```

This doc https://www.consul.io/docs/k8s/installation/overview walks through how to install the Helm chart, access Consul after it is deployed using the provisioned ACL tokens, and how to annotate pods to inject the sidecars. Also note the call out about service accounts in this doc https://www.consul.io/docs/k8s/connect/overview.

> The service name registered in Consul will be set to the name of the first container in the Pod. This can be customized with the `consul.hashicorp.com/connect-service` annotation. If using ACLs, this name **must** be the same as the Pod's ServiceAccount name.

See that same doc for example Pod or Deployment manifests.

Hope this helps. Let me know if you have any questions.

**Comment by mgfeller** [comment]
@blake thanks, that is really helpful! 

Yes, I'm looking for a basic configuration like this. 

I think the adapter needs some changes in order to support this configuration with ACLs enabled, I'll look into that and create an issue. 

In the meantime, I work on using a configuration similar to yours, with UI enabled, and without gossip encryption.

**Comment by mgfeller** [comment]
Fixed via #81
